src/components/sortingFunctions.js

```diff
@@ -0,0 +1,40 @@
+export const getShoppingItemBackgroundStyles = (
+  daysLeftForNextPurchase,
+  getDaysBetweenCurrentAndPreviousPurchase,
+  lastPurchasedOn,
+) => {
+  const presentDate = Date.now();
+  const daysBetweenCurrentAndPreviousPurchase = getDaysBetweenCurrentAndPreviousPurchase(
+    lastPurchasedOn,
+    presentDate,
+  );
+  if (daysBetweenCurrentAndPreviousPurchase >= daysLeftForNextPurchase * 2) {
+    return 'grey';
+  } else if (daysLeftForNextPurchase <= 7) {
+    return 'green';
+  } else if (daysLeftForNextPurchase > 7 && daysLeftForNextPurchase <= 30) {
+    return 'orange';
+  } else {
+    return 'red';
+  }
```

Comment:
ok I see. Yeah, the number situation was quite confusing in our latest discussions (on slack, as far as I saw). Since I always was changing the check/uncheck function by deleting the last two numbers of the `lastPurchasedOn` in firebase, the number of `daysLeftOfNextPurchase `was very high, so I thought using the app in a normal way would also produce a number higher than 30. This way of testing this function was not a great one, now I see how important it is to test it as realistically as possible. Thank you Luka for pointing this out.